### PR TITLE
fix: correctly select and use user locale and application title text [DHIS2-8525] [DHIS2-8723]

### PR DESCRIPTION
**Pull request by amcgee**
This does a few things, all related to the two localization issues with the headerbar ([DHIS2-8525](https://jira.dhis2.org/browse/DHIS2-8525) and [DHIS2-8723](https://jira.dhis2.org/browse/DHIS2-8723)

1. Update all `i18n.t` calls so that they are called **at render time, not at module initialization**
2. Remove the API call to `systemInfo` since it loads a bunch of security-sensitive information.  It was only used for (a) `contextPath` (which we can retrieve statically from the config context in app-runtime) and (b) `systemName` which is untranslated and replaced with `applicationTitle` below
3. Fetch `systemSettings/applicationTitle` which is the user-localized title for display in `headerBar`
4. Change `useEffect` to `useMemo`, to avoid needing a duplicate render (useEffect runs after first paint) and to avoid mutating the API response data (it should be treated as immutable)
5. Don't propagate `contextPath` throughout all components, instead loading the baseUrl from the config context in app-runtime wherever the baseUrl is actually needed.  Note that this is slightly brittle as-written since it makes the assumption that `baseUrl` ends in a `/` character.  We should probably change this to `joinPath(baseUrl, relativePath)` (a utility function we have in app-runtime but don't expose)
6. Ensure that we always set the locale to the user locale and the i18next namespace to `default`, even if we end up with a different singleton than the app uses (app-shell does this, so it should be a no-op in that environment).
7. Add a localization story with franglish strings
8. Update fixtures and cypress tests

I'd like to merge this to the `alpha` branch if it looks decent for testing in several apps, to avoid possible local dependency link issues.

### CODE REVIEW

File: src/HeaderBar/Apps.js [hunk 1]
```diff
@@ -66,7 +69,7 @@ function Search({ value, onChange, onIconClick, contextPath }) {
             </span>
 
             <span>
-                <a href={`${contextPath}/dhis-web-menu-management`}>
+                <a href={`${baseUrl}dhis-web-menu-management`}>
```
Comment (Mohammer5) [inline src/HeaderBar/Apps.js:72]: I would prefer to copy the `joinPath` helper for now (until exposed by app-runtime).
This seems too brittle and could break instances.

File: src/HeaderBar/Profile/ProfileHeader.js [hunk 2]
```diff
@@ -40,36 +41,36 @@ ProfileEmail.propTypes = {
...
-    <a
-        href={`${contextPath}/dhis-web-user-profile/#/profile`}
-        data-test="headerbar-profile-edit-profile-link"
-    >
-        {children}
+const ProfileEdit = ({ children }) => {
+    const baseUrl = useConfig().baseUrl
+    return (
+        <a
+            href={`${baseUrl}/dhis-web-user-profile/#/profile`}
```
Comment (Mohammer5) [inline src/HeaderBar/Profile/ProfileHeader.js]: Contrary to the `Apps.js`, this one adds a slash. I think I've seen it in other files also.
We could just copye the `joinPath` as mentioned above

File: cypress/integration/HeaderBar/The_HeaderBar_should_contain_a_logo_that_links_to_the_homepage/HeaderBar_contains_logo.js [hunk 3]
```diff
@@ -1,14 +1,13 @@
 import '../common/index.js'
 import { Then } from 'cypress-cucumber-preprocessor/steps'
+import { baseUrl } from '../common/index.js'
```
Comment (Mohammer5) [inline cypress/integration/HeaderBar/The_HeaderBar_should_contain_a_logo_that_links_to_the_homepage/HeaderBar_contains_logo.js:3]: I guess the import in line 3 should be enough and we can remove the one in line 1

File: src/HeaderBar/Profile/ProfileHeader.js (hunk 2 above)
Comment (amcgee) [reply src/HeaderBar/Profile/ProfileHeader.js]: Good catch, this is incorrect

File: src/HeaderBar/Apps.js (hunk 1 above)
Comment (amcgee) [reply src/HeaderBar/Apps.js:72]: Agreed, will do

**Pull request merged**